kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/KopTopic.java

```diff
@@ -106,15 +107,20 @@ public static String toString(String topic, int partition, String namespacePrefi
         return (new KopTopic(topic, namespacePrefix)).getPartitionName(partition);
     }
 
-    public static boolean isInternalTopic(final String fullTopicName) {
+    public static boolean isInternalTopic(final String fullTopicName, KafkaServiceConfiguration conf) {
         String partitionedTopicName = TopicName.get(fullTopicName).getPartitionedTopicName();
-        return partitionedTopicName.endsWith("/" + GROUP_METADATA_TOPIC_NAME)
-                || partitionedTopicName.endsWith("/" + TRANSACTION_STATE_TOPIC_NAME);
+        return partitionedTopicName.endsWith(conf.getKafkaMetadataNamespace() + "/" + GROUP_METADATA_TOPIC_NAME)
+                || partitionedTopicName.endsWith(conf.getKafkaMetadataNamespace() + "/" + TRANSACTION_STATE_TOPIC_NAME);
     }
 
-    public static boolean isGroupMetadataTopicName(final String fullTopicName) {
+    public static boolean isGroupMetadataTopicName(final String fullTopicName, KafkaServiceConfiguration conf) {
         String partitionedTopicName = TopicName.get(fullTopicName).getPartitionedTopicName();
-        return partitionedTopicName.endsWith("/" + GROUP_METADATA_TOPIC_NAME);
+        return partitionedTopicName.endsWith(conf.getKafkaMetadataNamespace() + "/" + GROUP_METADATA_TOPIC_NAME);
+    }
+
+    public static boolean isTransactionMetadataTopicName(final String fullTopicName, KafkaServiceConfiguration conf) {
+        String partitionedTopicName = TopicName.get(fullTopicName).getPartitionedTopicName();
+        return partitionedTopicName.endsWith(conf.getKafkaMetadataNamespace() + "/" + TRANSACTION_STATE_TOPIC_NAME);
     }
```

Comment:
Could you avoid passing the whole `KafkaServiceConfiguration` as the argument? I think here we only need a namespace name.

Since we've already used `TopicName` here, we can did the following check:

```java
    private static boolean isInternalTopic(final String fullTopicName,
                                           final String metadataNamespace,
                                           final Function<String, Boolean> condition) {
        final TopicName topicName = TopicName.get(fullTopicName);
        if (!topicName.getNamespace().equals(metadataNamespace)) {
            return false;
        }

        final String localName = topicName.getLocalName();
        return condition.apply(topicName.isPartitioned()
                ? localName.substring(0, localName.lastIndexOf(TopicName.PARTITIONED_TOPIC_SUFFIX))
                : localName);
    }

    public static boolean isInternalTopic(final String fullTopicName, final String metadataNamespace) {
        return isInternalTopic(fullTopicName, metadataNamespace,
                topic -> topic.equals(GROUP_METADATA_TOPIC_NAME) || topic.equals(TRANSACTION_STATE_TOPIC_NAME));
    }

    public static boolean isGroupMetadataTopicName(final String fullTopicName, final String metadataNamespace) {
        return isInternalTopic(fullTopicName, metadataNamespace, topic -> topic.equals(GROUP_METADATA_TOPIC_NAME));
    }
```

BTW, `isTransactionMetadataTopicName` is unused.

Reply:
I have addressed your comments,PTAL. Thanks.

Reply:
That's nothing. :)